packages/core/src/dialects/mssql/query-generator-internal.ts

```diff
@@ -45,4 +51,67 @@ export class MsSqlQueryGeneratorInternal<Dialect extends MssqlDialect = MssqlDia
 
     return fragment;
   }
+
+  getReturnFields(options: GetReturnFieldsOptions, modelDefinition?: ModelDefinition | null): string[] {
+    const returnFields: string[] = [];
+
+    if (Array.isArray(options.returning)) {
+      returnFields.push(...options.returning.map(field => {
+        if (typeof field === 'string') {
+          return this.queryGenerator.quoteIdentifier(field);
+        } else if (field instanceof Literal) {
+          // Due to how the mssql query is built, using a literal would never result in a properly formed query.
+          // It's better to warn early.
+          throw new Error(`literal() cannot be used in the "returning" option array in ${this.dialect.name}. Use col(), or a string instead.`);
```

Comment:
This check is a bit outdated. The logic should accept instances of `Col`, and strings, but ban all other values, not just `Literal`

Reply:
Updated the logic now